src/applications/gqrx/mainwindow.cpp

```diff
@@ -1185,16 +1187,20 @@ void MainWindow::selectDemod(int mode_idx)
     ui->plotter->setHiLowCutFrequencies(flo, fhi);
     ui->plotter->setClickResolution(click_res);
     ui->plotter->setFilterClickResolution(click_res);
-    rx->set_filter((double)flo, (double)fhi, d_filter_shape);
-    rx->set_cw_offset(cwofs);
-    rx->set_sql_level(uiDockRxOpt->currentSquelchLevel());
+    if (mode_idx != DockRxOpt::MODE_OFF)
+    {
+        rx->set_filter((double)flo, (double)fhi, d_filter_shape);
+        rx->set_cw_offset(cwofs);
+        rx->set_sql_level(uiDockRxOpt->currentSquelchLevel());
+     }
```

Comment:
Is this change related to the DCR/PLL changes, or is it fixing something else?

Reply:
This may be unrelated. It just saves 3 calls. When the demodulator is disabled there is no filter to set hicut/locut/transition/offset and no squelch too. So it's pointless to parameter of missing blocks.

I see another problem: AMPLL PLL bandwidth is not set on demod selection, but it should be set.

Reply:
Updated.